nomad/structs/services.go

```diff
@@ -1409,7 +1401,7 @@ func (p *ConsulProxy) Equal(o *ConsulProxy) bool {
 		return false
 	}
 
-	if !opaqueMapsEqual(p.Config, o.Config) {
```

Comment:
Is this safe? Can we be sure that Consul doesn't treat a nested empty map differently from a nil map? If the map is a nested value of Config (eg `Config["foo"] = map[string]string{}`), I could see nil meaning `unset` and empty meaning `set but use defaults`. Even it it's safe today can we be sure it will be safe for all future versions of Consul?

Maybe to put it another way: do we gain anything by not using DeepEqual in these cases? It seems like making _any_ assumptions about what's meaningful in the keys and values of opaque maps is ... well: not opaque!

Reply:
On the flip side - what if Consul normalizes these fields one way or the other, and the original job uses the other form? 

These opaque fields are a landmine no matter what, and I _think_ it's more likely we run into the normalization problem.